cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/ApplyLogic.java

```diff
@@ -84,227 +86,127 @@ public void setSpaceOperations(SpaceOperations spaceOperations) {
-        ConfigBean liveSpaceDevelopersConfig = createConfigFromSpaceDevelopers(liveSpaceDevelopers);
-
-        // compare entire configs as the diff wrapper is only suited for diff trees of
-        // these
-        log.info("Diffing space developers");
-        DiffResult wrappedDiff = this.diffLogic.createDiffResult(liveSpaceDevelopersConfig,
-            desiredSpaceDevelopersConfig);
-        log.verbose("Diffing space developers completed");
-
-        CfContainerChange spaceDevelopersChange = wrappedDiff.getSpaceDevelopersChange();
-        if (spaceDevelopersChange == null) {
-            log.info("No changes detected, nothing to apply");
-        } else {
-            Flux<Void> spaceDevelopersRequests = Flux.just(spaceDevelopersChange)
-                .flatMap(spaceDeveloperChange -> SpaceDevelopersRequestsPlanner
-                    .createSpaceDevelopersRequests(spaceDevelopersOperations, spaceDeveloperChange))
-                .onErrorContinue(log::warning);
-
-            log.info("Applying changes to space developers");
-
-            spaceDevelopersRequests.blockLast();
-            log.verbose("Changes to space developers applied");
-        }
+    public void setSpaceDevelopersOperations(SpaceDevelopersOperations spaceDevelopersOperations) {
+        this.spaceDevelopersOperations = spaceDevelopersOperations;
+    }
+
+    public void setServicesOperations(ServicesOperations servicesOperations) {
+        this.servicesOperations = servicesOperations;
+    }
+
+    public void setDiffLogic(DiffLogic diffLogic) {
+        this.diffLogic = diffLogic;
+    }
+
+    public void setGetLogic(GetLogic getLogic) {
+        this.getLogic = getLogic;
     }
 
     /**
-     * Apply the differences between the applications given in the yaml file and the
-     * configuration of the applications of your cf instance. In case of any
-     * non-recoverable error, the procedure is discontinued.
+     * Provides the service of manipulating the state of a cloud foundry instance
+     * such that it matches with a desired configuration ({@link ConfigBean}).
      *
-     * @param desiredApplications the applications that should all be present in the
-     *                            live system after the procedure
-     * @throws ApplyException       if an non-recoverable error occurs during the
-     *                              procedure
-     * @throws NullPointerException if the argument is null
+     * @param desiredConfigBean   desired configuration for a cloud foundry instance
+     * @param loginCommandOptions LoginCommandOptions
+     * @throws NullPointerException if one of the desired parameters is null.
      */
-    public void applyApplications(@Nonnull Map<String, ApplicationBean> desiredApplications) {
-        checkNotNull(desiredApplications);
+    public void apply(ConfigBean desiredConfigBean, OptionalLoginCommandOptions loginCommandOptions) {
```

Comment:
The `LoginCommandOptions` should not be used. Instead use the functions of `TargetOperations` within this method